web/html/src/manager/virtualization/guests/properties/guest-properties-form.js

```diff
@@ -0,0 +1,119 @@
+// @flow
+
+const React = require('react');
+const { Form } = require('components/input/Form');
+const { SubmitButton, Button } = require('components/buttons');
+const { Messages } = require('components/messages');
+
+declare function t(msg: string, ...args: Array<any>): string;
+
+type Props = {
+  submitText: string,
+  submit: Function,
+  initialModel: Object,
+  validationChecks: Array<{ check: (model: Object) => boolean, message: Object }>,
+  messages: Array<String>,
+  children: Function,
+};
+
+type State = {
+  model: Object,
+  isInvalid: boolean,
+  messages: Array<Object>,
+};
+
+/*
+ * Component editing a virtual machine properties
+ */
+class GuestPropertiesForm extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      model: Object.assign({}, props.initialModel),
+      isInvalid: false,
+      messages: [],
+    };
+  }
+
+  componentWillReceiveProps(nextProps: Props) {
```

Comment:
Didn't you migrate this to getderivedstatefromprops? Or is it on the other PR? This will probably not be called wit React 16 :)

Reply:
nice catch... it's a left over: I'll fix it